sql/core/src/main/scala/org/apache/spark/sql/streaming/Trigger.scala

```diff
@@ -38,6 +38,51 @@ sealed trait Trigger
 
 /**
  * :: Experimental ::
+ * A trigger that runs a query once then terminates
+ *
+ * Scala Example:
+ * {{{
+ *   df.write.trigger(OneTime)
+ * }}}
+ *
+ * Java Example:
+ * {{{
+ *   df.write.trigger(OneTime.create())
+ * }}}
+ *
+ * @since 2.2.0
+ */
+@Experimental
+@InterfaceStability.Evolving
+case class OneTime() extends Trigger
+
+/**
+ * :: Experimental ::
+ * Used to create [[OneTime]] triggers for [[StreamingQuery]]s.
```

Comment:
The explanation of OneTime trigger is given in the OneTime class definition. Does that suffice, or should the explanation be reiterated in the object definition?

If it should be reiterated, then I can do the same with ProcessingTime.